src/datasets/summarizations/trend-weekly-comparison-rate.summarization.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { queryFactory } from './trend-weekly-comparison-rate.summarization';
+import { TimeSeriesPoint } from '../metas/types';
+import { hasHighValidity, hasLowValidity, isTextPartsInSummary, norSummaryFilters } from './utils/tests';
+
+describe('queryFactory', () => {
+
+  it('should have validity value between 0 and 1.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 1; i <= 31; i++) {
+      const x = new Date(2020, 6, i);
+      const y = i;
+      points.push({ x, y });
+    }
+
+    const summaries = queryFactory(points)();
+    for (const summary of summaries) {
+      expect(summary.validity).toBeGreaterThanOrEqual(0.0);
+      expect(summary.validity).toBeLessThanOrEqual(1.0);
+    }
+  });
+
+  it('should create summaries describing monotonic linearly increasing data.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = i * 10;
+      points.push({ x, y });
+    }
+
+    const isFirstWeekSecondWeekSummary = isTextPartsInSummary('second week', 'first week', 'increased', 'in the same rate');

```

Comment:
Just curious, could we match a more complete sentence for these tests instead of 4 substrings?

Reply:
I want the test to be robust to small changes in summary templates. Also there might be some numerical values in the summaries.

Reply:
My only concern is that depending on what other words are in the summary, the meaning could be very different. I think that there are not so many tests that changing them each time the template changes would be too much of a hassle.